src/features/user/api/fetch-profile-image.ts

```diff
@@ -0,0 +1,25 @@
+import { ref, listAll, getDownloadURL } from "firebase/storage";
+import { imageDatabase } from "../../../shared/config/firebase-image";
+
+export const fetchProfileImage = async (id: number) => {
+  try {
+    const res = await listAll(ref(imageDatabase, "uploads"));
+
+    const promises = res.items.map(async (item) => {
+      const url = await getDownloadURL(item);
+      const originalString = item.fullPath;
+      const trimmedString = parseInt(originalString.substring("uploads/".length));
+      if (trimmedString === id) {
+        return url;
+      }
+    });
+
+    const urls = await Promise.all(promises);
+    const filteredUrls = urls.filter((url) => url !== undefined);
+
+    return filteredUrls[0]; // Zakładając, że istnieje tylko jedno pasujące URL do obrazu
```

Comment:
Dobrze, że zostawiłeś komentarz, bo tak to nie wiadomo o huj chodziło

Ale ogólnie komentarze po angielsku